apps/api/src/report/report.util.service.ts

```diff
@@ -119,6 +119,297 @@ export class ReportUtilService {
+          CASE WHEN COALESCE(ien_staus_hired.status_id, 0) > 0 THEN 1 ELSE 0 END as hired,
+          CASE WHEN COALESCE(ien_staus_withdrew.status_id, 0) > 0 THEN 1 ELSE 0 END as withdrawn
+        FROM public.ien_applicants as applicants
+        LEFT JOIN public.ien_applicant_status_audit ien_staus_hired 
+          ON ien_staus_hired.applicant_id=applicants.id AND ien_staus_hired.status_id=28 AND ien_staus_hired.start_date::date <= '${to}'
+        LEFT JOIN public.ien_applicant_status_audit ien_staus_withdrew 
+          ON ien_staus_withdrew.applicant_id=applicants.id AND ien_staus_withdrew.status_id=20 AND ien_staus_withdrew.start_date::date <= '${to}'
+        WHERE applicants.registration_date::date < '${to}'
+      ),
+      report AS (
+        SELECT 
+          id,
+          registration_date,
+          CASE WHEN hired = 0 AND withdrawn = 0 THEN 1 ELSE 0 END AS active,
+          CASE WHEN hired = 1 AND withdrawn = 0 THEN 1 ELSE 0 END AS hired,
+          CASE WHEN withdrawn = 1 THEN 1 ELSE 0 END AS withdrawn
+        FROM old_applicants
+      )
+      SELECT 
+        'Applied before ' || to_char('${from}'::date, 'Mon DD, YYYY') || ' (OLD)' AS title,
+        sum(active) AS active, sum(withdrawn) AS withdrawn, sum(hired) AS hired, count(*) AS total
+        FROM report WHERE registration_date::date < '${from}'
+      UNION ALL
+      SELECT 
+        'Applied after ' || to_char('${from}'::date, 'Mon DD, YYYY') || ' (NEW)' AS title,
+        sum(active) AS active, sum(withdrawn) AS withdrawn, sum(hired) AS hired, count(*) AS total
+        FROM report WHERE registration_date::date >= '${from}';
+    `;
+  }
+
+  licensingStageApplicantsQuery(from: string, to: string) {
+    return `
+      --col01 - 'Applied to NNAS' status: 4, 5, 6
+      --col02 - 'Applied to BCCNM' status: 7
+      --col03 - 'Completed English Language Requirement' status: 8
+      --col04 - 'Referred to NCAS' status: 9,10,11
+      --col05 - 'Completed NCAS (BCCNM Assessment)' status: 12
+      --col06 - 'Completed Additional Education' status: 13, 14
+      --col07 - 'Referred to NCLEX' status: 15,17
+      --col08 - 'NCLEX - Passed' status: 18
+      --col09 - 'BCCNM Licensed - Full License' status: 19
+      --col10 - 'Granted full licensure' status: 19
+      --col11 - 'Granted provisional licensure' status: 16
+      WITH active_applicants AS (
+        SELECT
+          t1.id,
+          COALESCE((SELECT max(sa.status_id) FROM public.ien_applicant_status_audit sa 
+            WHERE sa.applicant_id=t1.id AND sa.status_id IN (19, 16)), 0) AS has_license
+        FROM (
+          SELECT 
+            applicants.id,
+            CASE WHEN COALESCE(ien_staus_hired.status_id, 0) > 0 THEN 1 ELSE 0 END as hired,
+            CASE WHEN COALESCE(ien_staus_withdrew.status_id, 0) > 0 THEN 1 ELSE 0 END as withdrawn
+          FROM public.ien_applicants as applicants
+          LEFT JOIN public.ien_applicant_status_audit ien_staus_hired 
```

Comment:
Isn't it same one join with two conditions combined by 'or'?

Reply:
@sunghwan-park-fw
No, its not same as you may receive both status in some case. In such case we have to run group by and give priority to withdrew first in selection.

Use 2 instance of same table to find each detail separately help us to keep only one raw for an applicant. And we do not need to run a process of group by, partitions and provide row number based on order withdrawal then hired and pic row 0 for the same.